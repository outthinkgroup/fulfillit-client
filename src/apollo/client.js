import ApolloClient from "apollo-boost"
import fetch from "isomorphic-fetch"

export const isBrowser = () => typeof window !== "undefined"
const token = isBrowser() && window.localStorage.getItem("token")
console.log(token)
export const client = new ApolloClient({
  //uri: `https://fulfillit.mightyoak.co/graphql`, //?this needs to be an env var
  uri: process.env.GATSBY_WP_URL,
  fetch,
  request: operation => {
    const token = localStorage.getItem("token")
    operation.setContext({
      headers: {
        authorization: token ? `Bearer ${token}` : "",
      },
    })
  },
})
