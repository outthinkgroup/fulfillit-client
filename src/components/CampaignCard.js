import React, { useState, useContext } from "react"
import styled from "styled-components"
import { Link, navigate } from "gatsby"

import Card from "../elements/Card"
import Icon from "../elements/Icon"
import EditCampaign from "./EditCampaign"
import { LocalContext } from "../utils/LocalContext"

const CampaignCard = ({
  className,
  id,
  status,
  date,
  email,
  campaignOptions,
}) => {
  const [showActions, setShowActions] = useState(false)
  const dateFormated = `${new Date(date).getMonth() + 1}/${new Date(
    date
  ).getDate()}/${new Date(date).getFullYear()}`
  const emailAddress = `${email}@mg.taskcannon.co`
  const { localState, setLocalState } = useContext(LocalContext)
  return (
    <Card depth="low" className={className}>
      <div
        onMouseEnter={() => setShowActions(true)}
        onMouseLeave={() => setShowActions(false)}
      >
        <div className="top-section">
          <div className="title">
            <p className="tag">{status}</p>
            <h3>{campaignOptions.name}</h3>
            <p>{emailAddress}</p>
          </div>
          <div className="actions-group">
            {showActions && (
              <div className="actions">
                <button
                  type="button"
                  onClick={() => {
                    navigate(`/dashboard?campaign_id=${id}`)
                    setLocalState({ ...localState, isSideBarOpen: true })
                  }}
                  className="edit"
                  title="Edit"
                >
                  <Icon name="edit" color="white" />
                </button>
                <span className="analytics" title="Analytics">
                  <Icon name="chart" color="white" />
                </span>
                <span className="delete" title="Delete Campaign">
                  <Icon name="delete" color="white" />
                </span>
              </div>
            )}
          </div>
        </div>
        <div className="info">
          <div className="info-section">
            <p>Date Created</p>
            <span className="value">{dateFormated}</span>
          </div>
          <div className="info-section">
            <p>Total Transactions</p>
            <span className="value">{0}</span>
          </div>
        </div>
      </div>
    </Card>
  )
}

export default styled(CampaignCard)`
  padding: 0;
  color: black;
  background: #fff;
  box-shadow: ${({ theme }) => theme.depth.contrastLow};
  border-left: 4px solid ${({ theme }) => theme.colors.darkBlue};
  &:hover {
    box-shadow: ${({ theme }) => theme.depth.medium};
  }
  & > * {
    margin: 0;
  }
  .top-section {
    z-index: 2;
    padding: 10px 20px 10px;
    display: flex;
    justify-content: space-between;
  }
  .title {
    h3 {
      font-size: 28px;
      font-weight: 200;
      margin-top: 5px;
      text-transform: capitalize;
      color: ${({ theme }) => theme.colors.darkBlue};
    }
  }
  .info {
    color: ${({ theme }) => theme.colors.darkBlue};
    padding: 10px 20px;
    display: flex;
    background: ${({ theme }) => theme.colors.lightGrey};
    & > .info-section {
      width: 100%;
      text-align: center;
      &:not(:last-child) {
        border-right: 1px solid ${({ theme }) => theme.colors.lightBlue};
      }
    }
    p {
      margin-top: 0;
      font-weight: 500;
      font-size: 14px;
    }
    .value {
      font-size: 18px;
    }
  }

  .tag {
    margin: 0;
    display: inline-block;
    background: ${({ theme }) => theme.colors.lightGrey};
    color: ${({ theme }) => theme.colors.primary};
    font-size: 12px;
    letter-spacing: 1px;
    font-weight: 100;
    text-transform: uppercase;
    padding: 2px 4px;
  }

  .actions-group {
    position: absolute;
    top: 0;
    right: 0;

    svg {
      width: 20px;
    }
    .actions {
      box-shadow: inset 0 0px 25px rgba(50, 50, 93, 0.025),
        inset 0 0px 8px rgba(50, 50, 93, 0.07);
      font-size: 14px;
      display: flex;

      & > * {
        padding: 5px 8px;
        display: block;
        width: 100%;
        &:hover {
          font-weight: 700;
        }
      }
    }
  }
  .edit {
    background: ${({ theme }) => theme.colors.primary};
  }
  .analytics {
    background: ${({ theme }) => theme.colors.success.primary};
  }
  .delete {
    background: ${({ theme }) => theme.colors.warning.primary};
  }
  .action-icon {
    padding: 0;
    border-radius: 50%;
    width: 35px;
    height: 35px;
    background: #ddd;
    color: #999;
    display: flex;
    justify-content: center;
    align-items: center;
    box-shadow: none;
  }
`
