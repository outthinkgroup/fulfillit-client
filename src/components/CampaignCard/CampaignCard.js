import React, { useState, useContext } from "react"
import styled from "styled-components"
import { Link, navigate } from "gatsby"
import { useMutation } from "@apollo/react-hooks"
import { gql } from "apollo-boost"
import Card from "../../elements/Card"
import Icon from "../../elements/Icon"
import EditCampaign from "../EditCampaign/EditCampaign"
import { CAMPAIGNS } from "../CampaignList/CampaignList"
import { LocalContext } from "../../utils/LocalContext"

export const DELETE_CAMPAIGN = gql`
  mutation DELETE_CAMPAIGN($id: ID!) {
    deleteCampaign(input: { clientMutationId: "sdfsdaa", id: $id }) {
      campaign {
        id
      }
    }
  }
`

const CampaignCard = ({
  className,
  id,
  status,
  date,
  email,
  campaignOptions,
}) => {
  const [showActions, setShowActions] = useState(false)
  const dateFormated = `${new Date(date).getMonth() + 1}/${new Date(
    date
  ).getDate()}/${new Date(date).getFullYear()}`
  const emailAddress = `${email}@sendmagnet.com`
  const { localState, setLocalState } = useContext(LocalContext)
  const [
    deleteCampaign,
    { data: deletedData, loading: deleting, error: deleteError },
  ] = useMutation(DELETE_CAMPAIGN, {
    variables: {
      id,
    },
    update(cache, payload) {
      const data = cache.readQuery({ query: CAMPAIGNS })
      const { campaigns } = data.viewer
      console.log(payload)
      const { id } = payload.data.deleteCampaign.campaign
      data.viewer.campaigns = campaigns.nodes.filter(
        campaign => campaign.id !== id
      )
      cache.writeQuery({ query: CAMPAIGNS, data })
    },
    refetchQueries: ["CAMPAIGNS"],
  })
  return (
    <Card depth="low" className={className}>
      <div onMouseLeave={() => setShowActions(false)}>
        <div className="top-section">
          <div className="title">
            <p className="tag">{status}</p>
            <h3>{campaignOptions.name}</h3>
            <p>{emailAddress}</p>
          </div>
          <div className="actions-group">
            {showActions && (
              <div className="actions">
                <button
                  type="button"
                  onClick={() => {
                    navigate(`/dashboard?campaign_id=${id}`)
                    setLocalState({
                      ...localState,
                      isSideBarOpen: "EDIT_CAMPAIGN",
                    })
                  }}
                  className="edit"
                  title="Edit"
                >
                  <Icon name="edit" color="white" />
                </button>
                <button
                  type="button"
                  onClick={deleteCampaign}
                  className="analytics"
                  title="Analytics"
                >
                  <Icon name="chart" color="white" />
                </button>
                <button
                  type="button"
                  onClick={deleteCampaign}
                  className="delete"
                  title="Delete Campaign"
                >
                  <Icon name="delete" color="white" />
                </button>
              </div>
            )}
            <button
              type="button"
              onClick={e => {
                e.preventDefault()
                setShowActions(!showActions)
              }}
              className="options"
              title="Delete Campaign"
            >
              <Icon name="options" />
            </button>
          </div>
        </div>
        <div className="info">
          <div className="info-section">
            <p>Date Created</p>
            <span className="value">{dateFormated}</span>
          </div>
          <div className="info-section">
            <p>Total Transactions</p>
            <span className="value">{0}</span>
          </div>
        </div>
      </div>
    </Card>
  )
}

export default styled(CampaignCard)`
  padding: 0;
  color: black;
  background: #fff;
  max-width: 100%;
  overflow-x: hidden;
  box-shadow: ${({ theme }) => theme.depth.contrastLow};
  border-left: 4px solid ${({ theme }) => theme.colors.darkBlue};
  &:hover {
    box-shadow: ${({ theme }) => theme.depth.medium};
  }
  & > * {
    margin: 0;
  }
  .top-section {
    z-index: 2;
    padding: 10px 20px 10px;
    display: flex;
    justify-content: space-between;
  }
  .title {
    width: 100%;
    overflow-x: scroll;
    h3 {
      font-size: 28px;
      font-weight: 200;
      margin-top: 5px;
      text-transform: capitalize;
      color: ${({ theme }) => theme.colors.darkBlue};
    }
  }
  .info {
    color: ${({ theme }) => theme.colors.darkBlue};
    padding: 10px 20px;
    display: flex;
    background: ${({ theme }) => theme.colors.lightGrey};
    & > .info-section {
      width: 100%;
      text-align: center;
      &:not(:last-child) {
        border-right: 1px solid ${({ theme }) => theme.colors.lightBlue};
      }
    }
    p {
      margin-top: 0;
      font-weight: 500;
      font-size: 14px;
    }
    .value {
      font-size: 18px;
    }
  }

  .tag {
    margin: 0;
    display: inline-block;
    background: ${({ theme }) => theme.colors.lightGrey};
    color: ${({ theme }) => theme.colors.primary};
    font-size: 12px;
    letter-spacing: 1px;
    font-weight: 100;
    text-transform: uppercase;
    padding: 2px 4px;
  }

  .actions-group {
    position: absolute;
    top: 3px;
    right: 3px;
    display: flex;
    justify-content: flex-end;
    svg {
      width: 20px;
    }
    .actions {
      font-size: 14px;
      display: flex;

      & > * {
        height: 100%;
        border-radius: 50%;
        padding: 5px 8px;
        display: block;
        width: 35px;
        margin-left: 5px;
        display: flex;
        justify-content: center;
        align-items: center;
        height: 35px;
        box-shadow: ${({ theme }) => theme.depth.contrastLow};
        &:hover {
          font-weight: 700;
          transform: scale(1.1);
          box-shadow: ${({ theme }) => theme.depth.high};
        }
      }
    }
  }
  .edit {
    background: ${({ theme }) => theme.colors.primary};
  }
  .analytics {
    background: ${({ theme }) => theme.colors.success.primary};
  }
  .delete {
    background: ${({ theme }) => theme.colors.warning.primary};
  }
  .options {
    ${({ theme }) => theme.above.medium`
    
    `}
    background: transparent;
    box-shadow: none;
    &:hover,
    &:active,
    &:focus {
      background: #efefef;
    }
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 35px;
    height: 35px;
    padding: 0;
    margin-left: 5px;
  }
  .action-icon {
    padding: 0;
    border-radius: 50%;
    width: 35px;
    height: 35px;
    background: #ddd;
    color: #999;
    display: flex;
    justify-content: center;
    align-items: center;
    box-shadow: none;
  }
`
