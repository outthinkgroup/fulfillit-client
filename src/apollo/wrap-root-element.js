import React from "react"
import { ApolloProvider } from "@apollo/react-hooks"
import { client } from "./client"
import { ThemeProvider } from "styled-components"
import theme from "../designSystem/theme"

export const wrapRootElement = ({ element }) => (
  <ApolloProvider client={client}>
    <ThemeProvider theme={theme}>{element}</ThemeProvider>
  </ApolloProvider>
)
