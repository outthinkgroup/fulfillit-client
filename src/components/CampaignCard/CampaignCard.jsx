import React from "react";
import { gql } from "@apollo/client";
import { Link } from "react-router-dom";
import Card from "../../elements/Card";

const CampaignCard = ({ id, status, date, email, transactionCount, meta }) => {
  const dateFormatted = `${new Date(date).getMonth() + 1}/${new Date(
    date
  ).getDate()}/${new Date(date).getFullYear()}`;
  const emailAddress = `${email}@sendmagnet.com`;
  return (
    <Card
      depth="low"
      className="max-w-full overflow-hidden border-l-4 border-blue-900 bg-white p-0 text-black shadow shadow-blue-300 hover:shadow-lg"
    >
      <Link to={`/campaign/?campaign_id=${id}`}>
        <div className="z-10 flex justify-between py-3 px-6">
          <div className="w-full overflow-x-scroll">
            <p className="m-0 inline-block rounded bg-blue-100 p-1 text-xs uppercase tracking-wide text-blue-600">
              {status}
            </p>
            <h3 className="mt-1 text-2xl font-light capitalize text-blue-900">
              {meta.name}
            </h3>
            {/* TODO: get this working*/}
            <span data-link={`mailto:${emailAddress}`}>{emailAddress}</span>
          </div>
        </div>
        <div className="flex gap-4 bg-blue-50 py-3 px-6 text-blue-900">
          <div className="w-full">
            <p className="mt-0 text-sm font-bold">Date Created</p>
            <span className="text-base">{dateFormatted}</span>
          </div>
          <div className="w-full">
            <p className="mt-0 text-sm font-bold">Total Transactions</p>
            <span className="text-base">{transactionCount}</span>
          </div>
        </div>
      </Link>
    </Card>
  );
};
export default CampaignCard;
