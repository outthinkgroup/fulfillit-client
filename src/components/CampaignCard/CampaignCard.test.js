import React from "react"

import renderer from "react-test-renderer"
import CampaignCard from "./CampaignCard"
import { render, fireEvent, waitFor, screen } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { TestWrapper } from "../../apollo/wrap-root-element"

describe("CampaignCard", () => {
  let props = {
    id: "2",
    status: "Publish",
    date: "2020-12-15T18:32:55",
    email: "ronlieber",
    campaignOptions: {
      description: null,
      emailMarketingService: "mailchimp",
      name: "College Pre-order receipts",
    },
  }

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <CampaignCard {...props} />
        </TestWrapper>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("shows options", () => {
    const Comp = () => (
      <TestWrapper>
        <CampaignCard {...props} />
      </TestWrapper>
    )

    render(<Comp />)
    fireEvent.click(screen.getByTestId("option-btn"))

    expect(screen.getByTestId("actions-wrapper").childElementCount).toBe(3)
  })
})
