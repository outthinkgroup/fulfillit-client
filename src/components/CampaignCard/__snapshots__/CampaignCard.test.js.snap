// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignCard renders correctly 1`] = `
<div
  className="sc-bdVaJa dlXhdy sc-htpNat bFWwNG"
>
  <div
    onMouseLeave={[Function]}
  >
    <div
      className="top-section"
    >
      <div
        className="title"
      >
        <p
          className="tag"
        >
          Publish
        </p>
        <h3>
          College Pre-order receipts
        </h3>
        <p>
          ronlieber@sendmagnet.com
        </p>
      </div>
      <div
        className="actions-group"
      >
        <button
          className="options"
          data-testid="option-btn"
          onClick={[Function]}
          title="Delete Campaign"
          type="button"
        >
          <svg
            fill="black"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 12a2 2 0 1 1 0-4 2 2 0 0 1 0 4zm0-6a2 2 0 1 1 0-4 2 2 0 0 1 0 4zm0 12a2 2 0 1 1 0-4 2 2 0 0 1 0 4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="info"
    >
      <div
        className="info-section"
      >
        <p>
          Date Created
        </p>
        <span
          className="value"
        >
          12/15/2020
        </span>
      </div>
      <div
        className="info-section"
      >
        <p>
          Total Transactions
        </p>
        <span
          className="value"
        >
          0
        </span>
      </div>
    </div>
  </div>
</div>
`;
